src/WtsApi32.Desktop/WtsApi32+WtsSafeSessionInfoGuard.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Exported functions from the WtsApi32.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class WtsApi32
+    {
+        /// <summary>
+        /// Guard for Session info retreived from <see cref="WTSEnumerateSessions(IntPtr, int, int, ref WtsSafeSessionInfoGuard, ref int)"/>
+        /// </summary>
+        public partial class WtsSafeSessionInfoGuard : WtsSafeMemoryGuard, IEnumerable<WTS_SESSION_INFO>
```

Comment:
I don't think our SafeHandle's should be enumerable. They are meant as very low level wrappers to native handles -- not primary APIs to invoke other functionality.